src/layerstack.stanza

```diff
@@ -292,6 +316,49 @@ public defn LayerStack (
   ) -> LayerStack:
   #LayerStack(name, description, to-vector<LayerSpec>(initial-layers))
 
+defmethod write (o:OutputStream, s:LayerStack) :
+  println(o, "LayerStack: %_" % [value?(name(s))])
+  println(o, "  Layers: %_ copper layers %_ total layers" % [get-conductor-count(s) length(layers(s))])
+  for [idx, l] in enumerate(layers(s)) do:
+    println(o, "    %_ : %_ | %_ | %_" %
+      [idx, value?(name(l)) get-type(material(l)) thickness(l)])
+
+doc:\<DOC>
+  Usage: for [idx, line] in enumerate(lines) do: 
+           println(“Line %: %_” % [idx, line])
+<DOC>
+defn enumerate<?T> (s:Seqable<?T>) -> Seq<[Int, T]> :
+  var i = 0
+  for x in s seq :
+    next where:
+      val next = [i, x]
+      i = i + 1
+
+doc:\<DOC>
+  Get a layer by index
+  Usage: stack[idx]
+<DOC>
+public defn get (stack:LayerStack, idx:Int) -> LayerSpec :
+  layers(stack)[idx]
+
+doc:\<DOC>
+  Modify a layer by index
+  Usage: stack[idx] - new-layer
+<DOC>
+public defn set (stack:LayerStack, idx:Int, layer:LayerSpec) -> False :
```

Comment:
Why do we need this ? 
What is the use case that we want to add a `set` operator here ?

Reply:
I just thought we might need to support `set` since we support `get`.
This would replace a layer in the stack by another layer.
If you don't think that's useful, happy to remove it.

```
deftest(layerstack) test-set :
  #EXPECT(test-stack[0] == copper-3oz)
  test-stack[0] = copper-1oz
  #EXPECT(test-stack[0] == copper-1oz)

  #EXPECT(test-stack[1] == core-FR4-2)
  test-stack[1] = core-FR4-1
  #EXPECT(test-stack[1] == core-FR4-1)
```

Reply:
I want to avoid this kind of mutability. The goal is that the layer stack is declarative as much as possible.

Reply:
I'll remove the set function then.